runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsPlugin.java

```diff
@@ -233,27 +235,47 @@ public void onClientTick(ClientTick clientTick)
 				}
 
 				String oldTarget = entry.getTarget();
-				String newTarget = decorateTarget(oldTarget, decorations);
+				String newTarget = decorateTarget(oldTarget, player);
 
 				entry.setTarget(newTarget);
 			}
 		}
 	}
 
-	private String decorateTarget(String oldTarget, PlayerIndicatorsService.Decorations decorations)
+	private String decorateTarget(String oldTarget, Player player)
 	{
-		String newTarget = oldTarget;
+		PlayerIndicatorsService.Decorations decorations = playerIndicatorsService.getDecorations(player);
+
+		// Create a regex to find index of the col tag that precedes the player (if it exists).
+		// All spaces in the player name need to be replaced with \\p{Zs}.
```

Comment:
`\\p{Zs}` matches all whitespace characters, which I could add a comment explaining. 

It does not just add an extra backslash. The first parameter in replaceAll is a regex and the second parameter is the replacement - so it replaces the space(s) with the literal string in the second paramter. We need to add this to the regex since, iirc, the player name spaces have "regular" spaces and the entries in the menu has nbsp (\u00A0) in the player name. 

So, what the code below does is creates a regex string where the spaces in the player name are replaced with `\p{Zs}` to match the occurrences of nbsp in the menu entry string. 

So if we have the following name: `My Name`, the the regex created is: `My\p{Zs}Name`.